### PR TITLE
Feature/pp backend

### PR DESCRIPTION
**Pull request by Kapestian**
This pull request introduces new functionalities for profile picture management in the app and the Firebase Storage (cloud). It includes the addition of a new repository for handling profile pictures, updates to the ViewModel to incorporate this repository, and corresponding updates to the tests.

### Profile Picture Management:

* Added `ProfilePicRepository` interface and its implementation `ProfilePicRepositoryStorage` to manage profile pictures using Firebase Storage. (`ProfilePicRepository.kt`, `ProfilePicRepositoryStorage.kt`) [[1]](diffhunk://#diff-dd19e2423c35a3e579757bac48d3cd1235d25696b4e0e3d54fe6d020f609be7eR1-R18) [[2]](diffhunk://#diff-871a6990a978322aa0412d5260588bdeb21b0c69632f8c8521add06283ad7172R1-R99)

### ViewModel Updates:

* Updated `ProfilesViewModel` to include methods for uploading and deleting profile pictures, and modified its constructor to accept the new `ProfilePicRepository`. (`ProfilesViewModel.kt`) [[1]](diffhunk://#diff-4a28db3315acbc7e70a871270a4ea42b4c2d61d9dd4af0fa314981600305fc03R8-R16) [[2]](diffhunk://#diff-4a28db3315acbc7e70a871270a4ea42b4c2d61d9dd4af0fa314981600305fc03L35-R43) [[3]](diffhunk://#diff-4a28db3315acbc7e70a871270a4ea42b4c2d61d9dd4af0fa314981600305fc03R134-R168)

### Dependency Management:

* Added Firebase Storage dependency to the project. (`app/build.gradle.kts`, `gradle/libs.versions.toml`) [[1]](diffhunk://#diff-8cff73265af19c059547b76aca8882cbaa3209291406f52df1dafbbc78e80c46R156) [[2]](diffhunk://#diff-697f70cdd88ba88fe77eebda60c7e143f6ad1286bca75017421e93ad84fb87dfR50) [[3]](diffhunk://#diff-697f70cdd88ba88fe77eebda60c7e143f6ad1286bca75017421e93ad84fb87dfR103)

### Testing:

* Updated `ProfilesViewModelTest` to include tests for the new profile picture management functionality. (`ProfilesViewModelTest.kt`) [[1]](diffhunk://#diff-774d252c6c8ad17ee46f69531a249202e9922706f26fe22e47158c3851f33b4bR8-R10) [[2]](diffhunk://#diff-774d252c6c8ad17ee46f69531a249202e9922706f26fe22e47158c3851f33b4bR21) [[3]](diffhunk://#diff-774d252c6c8ad17ee46f69531a249202e9922706f26fe22e47158c3851f33b4bL38-R42) [[4]](diffhunk://#diff-774d252c6c8ad17ee46f69531a249202e9922706f26fe22e47158c3851f33b4bL48-R59) [[5]](diffhunk://#diff-774d252c6c8ad17ee46f69531a249202e9922706f26fe22e47158c3851f33b4bR209-R305)

### CODE REVIEW

File: app/src/main/java/com/github/se/icebreakrr/model/profile/ProfilesViewModel.kt [hunk 1]
```diff
@@ -139,6 +147,41 @@ open class ProfilesViewModel(private val repository: ProfilesRepository) : ViewM
...
+   * @param imageData The byte array of the image file to be uploaded.
+   * @throws IllegalStateException if the user is not logged in.
+   */
+  fun uploadCurrentUserProfilePicture(imageData: ByteArray) {
+    val userId = selectedProfile.value?.uid ?: throw IllegalStateException("User not logged in")
+    ppRepository.uploadProfilePicture(
+        userId = userId,
+        imageData = imageData,
+        onSuccess = { url ->
+          _selectedProfile.update { selected -> selected?.copy(profilePictureUrl = url) }
```
Comment (SamSteu) [inline app/src/main/java/com/github/se/icebreakrr/model/profile/ProfilesViewModel.kt:166]: _selectedProfile can be the profile of someone else. You need to make sure that this function is only called when _selectedProfile is set to your profile. If it is the case it seems good for me. Just check for that :)

File: app/src/main/java/com/github/se/icebreakrr/model/profile/ProfilesViewModel.kt [hunk 2]
```diff
@@ -139,6 +147,41 @@ open class ProfilesViewModel(private val repository: ProfilesRepository) : ViewM
...
+  /**
+   * Deletes the current user's profile picture from the remote storage system.
+   *
+   * @throws IllegalStateException if the user is not logged in.
+   */
+  fun deleteCurrentUserProfilePicture() {
+    ppRepository.deleteProfilePicture(
+        userId = selectedProfile.value?.uid ?: throw IllegalStateException("User not logged in"),
+        onSuccess = {
+          _selectedProfile.update { currentProfile ->
```
Comment (SamSteu) [inline app/src/main/java/com/github/se/icebreakrr/model/profile/ProfilesViewModel.kt:180]: same here.

**Review by SamSteu — COMMENTED** [review]
If _selectedUser in ProfilesViewModel is always set to your profile when you want to change it's profile picture. LGTM

File: app/src/main/java/com/github/se/icebreakrr/model/profile/ProfilesViewModel.kt (hunk 1 above)
Comment (Kapestian) [reply app/src/main/java/com/github/se/icebreakrr/model/profile/ProfilesViewModel.kt:168]: in theory, when you select your profile for edition, _selectedProfile is updated to your profile and stays there until you select someone else's profile. So as the profile picture upload option is only available from the personal edit screen, there is no chance to edit someone else's profile.

In practice, It is not impossible that a bug make another user's profile appear in the personal edit screen and create the vulnerability you are concerned about.

So I'll make a double security check , but in the next PR where I'll implement image upload from edit screen.

**Review by SamSteu — APPROVED** [review]
LGTM

**Pull request merged**